geopandas/tests/test_op_output_types.py

```diff
@@ -235,3 +234,51 @@ def identity(x):
         return x
 
     assert_object(df[["geometry2"]].apply(identity, axis=1), GeoDataFrame, None, None)
+
+
+def test_expanddim_in_apply():
+    # https://github.com/geopandas/geopandas/pull/2296#issuecomment-1021966443
+    s = GeoSeries.from_xy([0, 1], [0, 1])
+    assert_object(s.apply(lambda x: pd.Series([x.x, x.y])), pd.DataFrame)
```

Comment:
```suggestion
    result = s.apply(lambda x: pd.Series([x.x, x.y]))
    assert_object(result, pd.DataFrame)
```

(defining the result on a separate line makes it a bit easier if you are debugging such a test, and might also be a bit more readable)